src/main/java/com/keniu/repositories/BookRepositoryImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.keniu.repositories;
+
+import com.keniu.exceptions.EntityNotFoundException;
+import com.keniu.models.Book;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Override
+    public Book save(Book book) {
+        EntityManager entityManager = null;
+        EntityTransaction entityTransaction = null;
+        try {
+            entityManager = entityManagerFactory.createEntityManager();
+            entityTransaction = entityManager.getTransaction();
+            entityManager.persist(book);
+            entityTransaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (entityTransaction != null) {
+                entityTransaction.rollback();
+            }
+            throw new EntityNotFoundException("Can't save book " + book);
```

Comment:
EntityNotFoundException is used in Service layer

```suggestion
            throw new DataProcessingException("Can't save book " + book);
```

Reply:
Changes have been made